include/boost/fusion/container/generation/make_set.hpp

```diff
@@ -27,16 +30,22 @@ namespace boost { namespace fusion
         template <typename ...T>
         struct make_set
         {
-            typedef set<typename detail::as_fusion_element<T>::type...> type;
+            typedef set<
+                typename detail::as_fusion_element<
+                    typename remove_const<
```

Comment:
This series of expressions has happened a few times for perfect forwarding (I think). Should this be its own metafunction somewhere? Or is it already?